opensrp-reveal/src/main/java/org/smartregister/reveal/fragment/BaseOfflineMapsFragment.java

```diff
@@ -0,0 +1,145 @@
+
+                    message = bundle.getString(MapboxOfflineDownloaderService.KEY_RESULT_MESSAGE);
+
+                    if (MapboxOfflineDownloaderService.SERVICE_ACTION_RESULT.FAILED.name().equals(resultStatus)) {
+                        handleFailureResponse();
+                    } else {
+                        handleSuccessResponse();
+                    }
+                }
+            } else {
+                Timber.i(TAG, "Broadcast message has null Extras");
+            }
+
+        }
+    }
+
+    public void handleFailureResponse() {
+        if (!TextUtils.isEmpty(message) &&
+                !message.contains("MapBox Tile Count limit exceeded")) {
+            displayError(R.id.download_map,  message);
+        }
+
+        if (serviceAction == MapboxOfflineDownloaderService.SERVICE_ACTION.DELETE_MAP && !TextUtils.isEmpty(message)) {
+            displayError(R.id.download_map, message);
+        }
+    }
+
+    public void handleSuccessResponse() {
+        // We should disable the stop offline download button if it was stopped successfully
+        if (serviceAction == MapboxOfflineDownloaderService.SERVICE_ACTION.STOP_CURRENT_DOWNLOAD) {
+            currentMapDownload = null;
+            downloadStopped(mapUniqueName);
+        } else if (serviceAction == MapboxOfflineDownloaderService.SERVICE_ACTION.DELETE_MAP) {
+            mapDeletedSuccessfully(mapUniqueName);
+        } else {
+            if (!TextUtils.isEmpty(message)) {
+                // This is a download progress message
+                if (isValidDouble(message)) {
+                    if (Double.valueOf(message) == 100d) {
+                        currentMapDownload = null;
+                        displayToast(getString(R.string.download_finished_successfuly));
+                        downloadCompleted(mapUniqueName);
+                        // setCanStopMapDownload(false);
+                    } else {
+                        // setCanStopMapDownload(true);
+                        displayToast(getString(R.string.map_download_progress, Double.valueOf(message)));
+                        downloadStarted(mapUniqueName);
+                    }
+                } else {
+                    displayToast(message);
+                }
+            }
+        }
+    }
+
+    protected boolean isValidDouble(String doubleString) {
+        String doubleRegex = "[+-]{0,1}[0-9]*.{0,1}[0-9]*";
+        return (!doubleString.isEmpty() && doubleString.matches(doubleRegex));
+    }
+
+    protected void downloadCompleted(String mapUniqueName) {
+        // Implement in child class
+    }
+
+    protected void downloadStarted(String mapUniqueName) {
+        // Implement in child class
+    }
+
+    protected void mapDeletedSuccessfully(String mapUniqueName) {
+        // implement in child class
+    }
+
+    protected void downloadStopped(String mapUniqueName) {
+        // implement in child class
+    }
```

Comment:
maybe then change this class to abstract so that the methods are implemented on child classes